group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2358,9 +2358,18 @@ private CoordinatorResult<StreamsGroupHeartbeatResult, CoordinatorRecord> stream
             || hasAssignedStandbyTasksChanged(member, updatedMember)
             || hasAssignedWarmupTasksChanged(member, updatedMember)
         ) {
+            streamsGroupMembersSendMetadata.clear();
             response.setActiveTasks(createStreamsGroupHeartbeatResponseTaskIds(updatedMember.assignedActiveTasks()));
             response.setStandbyTasks(createStreamsGroupHeartbeatResponseTaskIds(updatedMember.assignedStandbyTasks()));
             response.setWarmupTasks(createStreamsGroupHeartbeatResponseTaskIds(updatedMember.assignedWarmupTasks()));
+        } else {
+            long memberGroupEpochCount = group.members().values().stream().filter(m -> m.memberEpoch() == group.groupEpoch()).count();
+            if (memberGroupEpochCount == group.members().size()) {
+                if (!streamsGroupMembersSendMetadata.contains(updatedMember.memberId())) {
+                    response.setPartitionsByUserEndpoint(maybeBuildEndpointToPartitions(group));
+                    streamsGroupMembersSendMetadata.add(updatedMember.memberId());
+                }
```

Comment:
If all members have a `memberEpoch` equaling the `groupEpoch` and the `memberId` is not in the sent metadata set, add the endpoint information to the heartbeat response and add the id to the set to prevent redundant updates.